packages/firebase_admob/lib/firebase_admob.dart

```diff
@@ -233,7 +233,11 @@ abstract class MobileAd {
   /// Disposing a banner ad that's been shown removes it from the screen.
   /// Interstitial ads can't be programmatically removed from view.
   Future<bool> dispose() {
-    assert(_allAds[id] != null);
+    // When dispose is called, following assert prevents graceful cleanup of resources
+    // and occationally causes exception in the plugin which results in dangling
+    // BannerAd in the app. Hence commenting it for now till better way of handling is
+    // added by the plugin maintainer.
+    // assert(_allAds[id] != null);
```

Comment:
I think you could replace this line with something like

```
if (_allAds[id] == null) return Future<bool>.value(true);

Reply:
Thanks @bparrishMines for suggestion. I have commit changes at e0a82e1.